backend/utilities/cwe_map_generator/main.py

```diff
@@ -0,0 +1,111 @@
+"""
+Generates a JSON mapping of CWE IDs to names.
+
+The CWE database is published as an XML file within a zip archive.
+This downloads the database and extracts the CWE info.
+"""
+# pyright: strict
+
+import json
+import logging
+from pathlib import Path
+import requests
+from sys import stderr
+import tempfile
+import time
+from typing import Any, IO
+import xmltodict
+import zipfile
+
+LOG = logging.getLogger("cwe_map_generator")
+logging.basicConfig(stream=stderr, level=logging.INFO)
+
+CACHE_DIR = Path(".cache")
+CWE_DB_URL = "https://cwe.mitre.org/data/xml/cwec_latest.xml.zip"
+
+# Maps CWE ID to name.
+CWEMapping = dict[int, str]
+
+
+def fetch(src: str, dest: Path):
+    """Download / update the database."""
+    dest.parent.mkdir(mode=0o700, parents=True, exist_ok=True)
+
+    headers: dict[str, str] = {}
+    try:
+        headers["If-Modified-Since"] = time.strftime("%a, %d %b %Y %H:%M:%S GMT", time.gmtime(dest.stat().st_mtime))
+    except FileNotFoundError:
+        pass
+
+    LOG.info(f"Downloading {src} -> {dest}")
+
+    with requests.get(src, headers=headers, stream=True) as req:
+        if req.status_code == 304:
+            LOG.info("Database up-to-date.")
+            return
+        req.raise_for_status()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            outfile = Path(tempdir) / "download.tmp"
+            with outfile.open("wb") as f:
+                for chunk in req.iter_content(chunk_size=64 * 1024):
```

Comment:
Possibly a naive question, is there a special significance to this chunk size of 64 KiB that we should document in code comments?

Reply:
Nope, just an arbitrary buffer size.